src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ElementsMarshalling.cs

```diff
@@ -300,13 +418,34 @@ public override StatementSyntax GenerateMarshalStatement(TypePositionInfo info,
                             IdentifierName(nativeSpanIdentifier),
                             IdentifierName("Clear")))));
             }
+            if (MarshallerHelpers.MarshalsOutToLocal(info, context) && _isStateful)
+            {
+                var elementType = _parameterPointedToType;//info.IsByRef ? PointerType(_unmanagedElementType) : _unmanagedElementType;
+                // <value_native_out> = (<nativeType>*)Unsafe.AsPointer(ref <nativeSpan>.GetPinnableReference());
```

Comment:
Similar question here around using a "pinnable reference" without pinning. Are we sure that this works? If this is only in the "stateful element" case, we shouldn't add code that will do bad things and instead we should gracefully fail earlier.